### PR TITLE
Introduce version v1beta2 for Dynakube

### PR DESCRIPTION
**Pull request by waodim**
## Description

With this a new version `v1beta2` of the Dynakube is introduced.

Noteworthy changes:
- a new field `metaDataEnrichment` was added to the .spec of the Dynakube.  it has an enabled field (enabled by default) and a namespace selector.
- a new field `dynatraceApiRequestsThreshold` was added to the .spec of the Dynakube.
- a new field `secCompProfile` was added to relevant oneAgent sections: `hostMonitoring`, `classicFullStack`, `cloudNativeFullstack`

Please have a look at the tags if they are correct and also check my wording please.

Corresponding JIRA ticket: [K8S-9331](https://dt-rnd.atlassian.net/browse/K8S-9331)

## How can this be tested?

Not sure how this could be tested. Maybe run `make manifests` and check the dynakube CRD.

## Checklist

- [X] Unit tests have been updated/added
- [x] PR is labeled accordingly with a single label
- [x] I have read and understood the [contribution guidelines](https://github.com/Dynatrace/dynatrace-operator/blob/main/CONTRIBUTING.md)

### CODE REVIEW

**Review by 0sewa0 — DISMISSED** [review]

File: pkg/api/v1beta2/dynakube/dynakube_conversion.go [hunk 1]
```diff
@@ -0,0 +1,207 @@
+package dynakube
+
+import (
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/v1alpha1/dynakube"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+)
+
+// ConvertTo converts v1beta2 to v1beta1.
+func (src *DynaKube) ConvertTo(dstRaw conversion.Hub) error {
```
[the remark this answers is not in the document]
Comment (waodim) [reply pkg/api/v1beta2/dynakube/dynakube_conversion.go:10]: so basically I can remove the conversion.go files for now?

File: pkg/api/v1beta2/dynakube/oneagent_types.go [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+}
+
+type CloudNativeFullStackSpec struct {
+	AppInjectionSpec `json:",inline"`
+	HostInjectSpec   `json:",inline"`
+
+	// Applicable only for applicationMonitoring or cloudNativeFullStack configuration types. The namespaces where you want Dynatrace Operator to inject.
+	// For more information, see Configure monitoring for namespaces and pods (https://www.dynatrace.com/support/help/setup-and-configuration/setup-on-container-platforms/kubernetes/get-started-with-kubernetes-monitoring/dto-config-options-k8s#annotate).
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Namespace Selector",order=17,xDescriptors="urn:alm:descriptor:com.tectonic.ui:selector:core:v1:Namespace"
+	NamespaceSelector metav1.LabelSelector `json:"namespaceSelector,omitempty"`
```
[the remark this answers is not in the document]
Comment (waodim) [reply pkg/api/v1beta2/dynakube/oneagent_types.go:45]: done in https://github.com/Dynatrace/dynatrace-operator/pull/2998/commits/0e1815371557d63e4677930eaa39566050ec7beb

File: pkg/api/v1beta2/dynakube/oneagent_types.go [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+	HostMonitoring *HostInjectSpec `json:"hostMonitoring,omitempty"`
+
+	// Sets a host group for OneAgent.
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Host Group",order=5,xDescriptors={"urn:alm:descriptor:com.tectonic.ui:advanced","urn:alm:descriptor:com.tectonic.ui:text"}
+	HostGroup string `json:"hostGroup,omitempty"`
+
+	// The SecComp Profile that will be configured in order to run in secure computing mode.
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="OneAgent SecComp Profile",order=17,xDescriptors="urn:alm:descriptor:com.tectonic.ui:selector:core:v1:Namespace"
+	SecCompProfile string `json:"secCompProfile,omitempty"`
```
[the remark this answers is not in the document]
Comment (waodim) [reply pkg/api/v1beta2/dynakube/oneagent_types.go]: done in https://github.com/Dynatrace/dynatrace-operator/pull/2998/commits/0e1815371557d63e4677930eaa39566050ec7beb

File: pkg/api/v1beta2/dynakube/feature_flags.go [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+	AnnotationFeatureOneAgentInitialConnectRetry    = AnnotationFeaturePrefix + "oneagent-initial-connect-retry-ms"
+	AnnotationFeatureRunOneAgentContainerPrivileged = AnnotationFeaturePrefix + "oneagent-privileged"
+	AnnotationFeatureOneAgentSecCompProfile         = AnnotationFeaturePrefix + "oneagent-seccomp-profile"
+
+	// injection (webhook).
+
+	// Deprecated: AnnotationFeatureDisableMetadataEnrichment use AnnotationFeatureMetadataEnrichment instead.
+	AnnotationFeatureDisableMetadataEnrichment = AnnotationFeaturePrefix + "disable-metadata-enrichment"
+
+	AnnotationFeatureMetadataEnrichment = AnnotationFeaturePrefix + "metadata-enrichment"
```
[the remark this answers is not in the document]
Comment (waodim) [reply pkg/api/v1beta2/dynakube/feature_flags.go:63]: done in https://github.com/Dynatrace/dynatrace-operator/pull/2998/commits/0e1815371557d63e4677930eaa39566050ec7beb

File: pkg/api/v1beta2/dynakube/dynakube_conversion.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (waodim) [reply pkg/api/v1beta2/dynakube/dynakube_conversion.go:10]: done in https://github.com/Dynatrace/dynatrace-operator/pull/2998/commits/0e1815371557d63e4677930eaa39566050ec7beb

File: pkg/api/v1beta2/dynakube/deprecated_activegate_types.go [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+	Enabled bool `json:"enabled,omitempty"`
+}
+
+// Deprecated: Use ActiveGateSpec instead.
+type KubernetesMonitoringSpec struct {
+	CapabilityProperties `json:",inline"`
+	// Enables Capability
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Capability",order=29,xDescriptors="urn:alm:descriptor:com.tectonic.ui:selector:booleanSwitch"
+	Enabled bool `json:"enabled,omitempty"`
+}
```
[the remark this answers is not in the document]
Comment (waodim) [reply pkg/api/v1beta2/dynakube/deprecated_activegate_types.go]: done in https://github.com/Dynatrace/dynatrace-operator/pull/2998/commits/0e1815371557d63e4677930eaa39566050ec7beb

File: pkg/api/v1beta2/dynakube/oneagent_types.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (0sewa0) [reply pkg/api/v1beta2/dynakube/oneagent_types.go:45]: is it 🙃 😅 ?

File: pkg/api/v1beta2/dynakube/feature_flags.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (0sewa0) [reply pkg/api/v1beta2/dynakube/feature_flags.go:63]: is it 🙃 😅 ?

[the remark this answers is not in the document]
Comment (waodim) [reply pkg/api/v1beta2/dynakube/feature_flags.go]: sorry for that somehow missed it

should be done in https://github.com/Dynatrace/dynatrace-operator/pull/2998/commits/3b96f854b85d5313965341b2e25259c021484f25

File: doc/api/dynakube-api-ref.md [hunk 6]
```diff
@@ -40,6 +40,13 @@
 |tolerations|Set tolerations for the ActiveGate pods|-|array|
 |topologySpreadConstraints|Adds TopologySpreadConstraints for the ActiveGate pods|-|array|
 
+### .spec.metaDataEnrichment
+
+|Parameter|Description|Default value|Data type|
+|:-|:-|:-|:-|
+|enabled|Enables MetaDataEnrichment Enabled by default.|-|boolean|
```
Comment (aorcholski) [inline doc/api/dynakube-api-ref.md:47]: the default value is known:
```
|enabled|Enables MetaDataEnrichment Enabled by default.|true|boolean|
```

Comment (waodim) [reply doc/api/dynakube-api-ref.md]: done in https://github.com/Dynatrace/dynatrace-operator/pull/2998/commits/377ee7de25d0ef6e3aaa1218f47834a1536c95e2

**Review by aorcholski — APPROVED** [review]
LGTM

**Review by wepudt — APPROVED** [review]
lgtm

**Pull request merged**